src/Model/ModelManagerInterface.php

```diff
@@ -82,8 +82,8 @@ public function findBy($class, array $criteria = []);
     public function findOneBy($class, array $criteria = []);
 
     /**
-     * @param string $class
-     * @param mixed  $id
+     * @param string     $class
+     * @param int|string $id
```

Comment:
It is not possible to pass a `UserId` object or anything else here?

Reply:
1) All our implementations are considering that id is an int or a string
https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/3.x/src/Model/ModelManager.php#L271

2) We should restrict phpDoc to what we're need.
If you look at the code in SonataAdmin, all the occurence of 
```
$this->modelManager->find($class, $id)
```
Is using a int or string ; so we don't need the modelManager to support more.

Reply:
Do you know which is the use case for [this](https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/2bac2f267b27b944288ac671d69a321a491ea10b/tests/Model/ModelManagerTest.php#L567-L568)?
```php
$id = new ProductId(12345);
$entity = new Product($id, 'Some product');
```

Reply:
By the way, the method is
```
    /**
     * Get the identifiers of this model class.
     *
     * This returns an array to handle cases like a primary key that is
     * composed of multiple columns. If you need a string representation,
     * use getNormalizedIdentifier resp. getUrlSafeIdentifier
     *
     * @param object $model
     *
     * @return array list of all identifiers of this model
     */
    public function getIdentifierValues($model);
```

But the usage is
```
// ModelChoiceLoader or ModelToArrayTransformer
$id = implode(AdapterInterface::ID_SEPARATOR, $this->getIdentifierValues($model));
```

So I think we should also update the `getIdentifierValues` phpdoc to return `array<int|string>` or `string[]`.

Reply:
> It is not possible to pass a `UserId` object or anything else here?

I usually use [`UUIDs` objects ](https://github.com/ramsey/uuid)as ids.